### PR TITLE
Include documentation defintion files in pyodide releases

### PR DESCRIPTION
**Pull request by Hephaistos7**
### Description

Added Documentation to https://pyodide.org/en/stable/usage/downloading-and-deploying.html.

Proposed Solution to Issue #4942 #4945 .

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by Hephaistos7** [comment]
@hoodmane Feel free to give me feedback to the additions in the documentation and the changelog.

**Comment by hoodmane** [comment]
No changelog entry needed for this since it's a documentation-only change.

File: docs/usage/downloading-and-deploying.md [hunk 1]
```diff
@@ -56,3 +56,40 @@ services.
 For additional suggestions for optimizing the size and load time for Pyodide,
 see the [Emscripten documentation about
 deployments](https://emscripten.org/docs/compiling/Deploying-Pages.html).
+
+## Type Definitions of Pyodide files
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:60]:
```suggestion
## Contents of Pyodide github releases
```

File: docs/usage/downloading-and-deploying.md [hunk 2]
```diff
@@ -56,3 +56,40 @@ services.
...
+
+### Core Release Files
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:62]:
```suggestion
### Files in `pyodide-core-{{VERSION}}.tar.bz2`
```

File: docs/usage/downloading-and-deploying.md [hunk 3]
```diff
@@ -56,3 +56,40 @@ services.
...
+| pyodides.mjs      | A small JS loader shim which exports loadPyodide. It manages downloading the runtime and handling user settings.                                                                                                                                  |
+| python_stdlin.zip | The Python Standard Library for pyodide. A zip file consisting of the Python Lib folder (except a few things we've unvendored) and the Pyodide Python runtime libraries. Mounted directly into the Pyodide FS and imported using ZipImporter      |
+| package.json      | Tells node how to use Pyodide, since pyodide-core was primarily intended for use with node.                                                                                                                                                       |
+| pyodide-lock.json | Lockfile for Python packages, used by pyodide.loadPackage and micropip.install. Necessary in all cases.                                                                                                                                           |
+
+### Full Release Additional Files
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:74]:
```suggestion
### Additional Files in `pyodide-{{VERSION}}.tar.bz2`
```

**Comment by hoodmane** [comment]
Can you update the file list to reflect the changes in #4945? So `*.d.ts` will be in core files, and makesnap.mjs, snapshot.bin, module_test.html,  test.html, module_webworker_dev.js  webworker_dev.js have been deleted.

File: docs/usage/downloading-and-deploying.md [hunk 4]
```diff
@@ -56,3 +56,40 @@ services.
...
+| :---------------------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| fonts/                  | Used by matplotlib-pyodide.                                                                                                                                                                          |
+| ffi.d.ts                | Typescript type definitions, useful if you want to use Pyodide in a typescript project                                                                                                               |
+| pyodide.d.ts            | Typescript type definitions, useful if you want to use Pyodide in a typescript project                                                                                                               |
+| 227 \*.whl              | Contains various Python Wheels like NumPy, Pandas, SciPy, etc. When downloading and including in a project, these no longer need to be loaded from Pyodide's online wheel repository during runtime. |
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:81]: These numbers are unlikely to stay up to date... We might need a script to check this.

File: docs/usage/downloading-and-deploying.md (hunk 2 above)
Comment (Hephaistos7) [reply docs/usage/downloading-and-deploying.md]: Done!

File: docs/usage/downloading-and-deploying.md (hunk 1 above)
Comment (Hephaistos7) [reply docs/usage/downloading-and-deploying.md]: Done!

File: docs/usage/downloading-and-deploying.md (hunk 3 above)
Comment (Hephaistos7) [reply docs/usage/downloading-and-deploying.md]: Done!

File: docs/usage/downloading-and-deploying.md (hunk 4 above)
Comment (Hephaistos7) [reply docs/usage/downloading-and-deploying.md]: I removed the number of files. It is kind of unnecessary to include the exact number tbf.

File: docs/usage/downloading-and-deploying.md [hunk 5]
```diff
@@ -56,3 +56,35 @@ services.
...
+| pyodide.d.ts    | Typescript type definitions, useful if you want to use Pyodide in a typescript project.                                                                                                              |
+| \*.whl          | Contains various Python Wheels like NumPy, Pandas, SciPy, etc. When downloading and including in a project, these no longer need to be loaded from Pyodide's online wheel repository during runtime. |
+| \*.metadata     | Information about the included wheels, such as name, author, license, dependencies, etc. Allows more efficient package resolution, specified in [PEP 658](https://peps.python.org/pep-0658/).        |
+| \*.zip          | These are shared libraries and unvendored Python standard library modules.                                                                                                                           |
+|                 |
+| python          | Bash/node polyglot. Only needed for node < 18.                                                                                                                                                       |
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:85]:
```suggestion
| python          | Python CLI command that uses Pyodide as the interpreter  |
```

File: docs/usage/downloading-and-deploying.md [hunk 6]
```diff
@@ -56,3 +56,35 @@ services.
...
+| pyodide.asm.wasm  | The WebAssembly half of the main "binary". Direct output from the Emscripten compiler. Contains all C library code that is statically linked. Also includes enough of libcxx to support things like exceptions in extension modules that use C++. |
+| pyodide.js        | Exports loadPyodide on globalThis. Legacy support for people who can't use es6 modules for whatever reason. Prefer pyodide.mjs.                                                                                                                   |
+| pyodides.mjs      | A small JS loader shim which exports loadPyodide. It manages downloading the runtime and handling user settings.                                                                                                                                  |
+| python_stdlin.zip | The Python Standard Library for pyodide. A zip file consisting of the Python Lib folder (except a few things we've unvendored) and the Pyodide Python runtime libraries. Mounted directly into the Pyodide FS and imported using ZipImporter.     |
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:70]:
```suggestion
| python_stdlib.zip | The Python Standard Library for pyodide. A zip file consisting of the Python Lib folder (except a few things we've unvendored) and the Pyodide Python runtime libraries. Mounted directly into the Pyodide FS and imported using ZipImporter.     |
```

File: docs/usage/downloading-and-deploying.md [hunk 7]
```diff
@@ -56,3 +56,35 @@ services.
...
+| package.json      | Tells node how to use Pyodide, since pyodide-core was primarily intended for use with node.                                                                                                                                                       |
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:71]:
```suggestion
| package.json      | The node package manifest for Pyodide                                                                                                                                                        |
```

File: docs/usage/downloading-and-deploying.md [hunk 8]
```diff
@@ -56,3 +56,35 @@ services.
...
+| pyodide-lock.json | Lockfile for Python packages, used by pyodide.loadPackage and micropip.install. Necessary in all cases.                                                                                                                                           |
```
Comment (hoodmane) [inline docs/usage/downloading-and-deploying.md:72]:
```suggestion
| pyodide-lock.json | Lockfile for Python packages, used by `pyodide.loadPackage()` and `micropip.install()`.                                                                                                                                           |
```

**Comment by Hephaistos7** [comment]
Thank you for reviewing.

File: docs/usage/downloading-and-deploying.md [hunk 9]
```diff
@@ -56,3 +56,35 @@ services.
 For additional suggestions for optimizing the size and load time for Pyodide,
 see the [Emscripten documentation about
 deployments](https://emscripten.org/docs/compiling/Deploying-Pages.html).
+
+## Contents of Pyodide github releases
```
Comment (agriyakhetarpal) [inline docs/usage/downloading-and-deploying.md:60]:
```suggestion
## Contents of Pyodide GitHub releases
```

(nit)

File: docs/usage/downloading-and-deploying.md [hunk 10]
```diff
@@ -56,3 +56,35 @@ services.
...
+| pyodide.asm.js    | The JavaScript half of the main "binary". Direct output from the Emscripten compiler. Contains the Emscripten bootstrap code + all JavaScript libraries used by C + the JavaScript/Wasm runtime interop APIs.                                     |
+| pyodide.asm.wasm  | The WebAssembly half of the main "binary". Direct output from the Emscripten compiler. Contains all C library code that is statically linked. Also includes enough of libcxx to support things like exceptions in extension modules that use C++. |
+| pyodide.js        | Exports loadPyodide on globalThis. Legacy support for people who can't use es6 modules for whatever reason. Prefer pyodide.mjs.                                                                                                                   |
+| pyodides.mjs      | A small JS loader shim which exports loadPyodide. It manages downloading the runtime and handling user settings.                                                                                                                                  |
```
Comment (agriyakhetarpal) [inline docs/usage/downloading-and-deploying.md:69]:
```suggestion
| pyodide.mjs      | A small JS loader shim which exports loadPyodide. It manages downloading the runtime and handling user settings.                                                                                                                                  |
```

based on the description for `pyodide.js` – this should be `pyodide.mjs`

File: docs/usage/downloading-and-deploying.md (hunk 6 above)
Comment (agriyakhetarpal) [inline docs/usage/downloading-and-deploying.md:70]:
```suggestion
| python_stdlib.zip | The Python Standard Library for pyodide. A zip file consisting of the Python Lib folder (except a few things we've unvendored) and the Pyodide Python runtime libraries. Mounted directly into the Pyodide FS and imported using ZipImporter.     |
```

**Comment by Hephaistos7** [comment]
Thank you @agriyakhetarpal :)

**Comment by hoodmane** [comment]
Should we add a file called `CONTENTS.md` or something like this to the release with a link to this section in the docs? Because one problem with this is that if someone is wondering what is in the release it's not obvious how they would make it to this description.

**Comment by agriyakhetarpal** [comment]
That would be great! We don't have any text in the GitHub Release to show up in the release notes, it might be good to add something there with links for easy access.

**Comment by hoodmane** [comment]
I guess we should merge this as is and then add a cross reference so that people can discover it in a followup. Thanks @Hephaistos7!

**Pull request merged**